src/naksu/config/config.go

```diff
@@ -0,0 +1,108 @@
+package config
+
+import (
+	"fmt"
+	"naksu/mebroutines"
+	"strconv"
+
+	"github.com/go-ini/ini"
+)
+
+var cfg *ini.File
+
+func setIfMissing(section string, key string, defaultValue string) {
+	if !cfg.Section(section).HasKey(key) {
+		cfg.Section(section).Key(key).SetValue(defaultValue)
+	}
+}
+
+type defaultValue struct {
+	section, key, value string
+}
+
+func fillDefaults() {
+	defaults := []defaultValue{
+		defaultValue{"common", "iniVersion", strconv.FormatInt(1, 10)},
+		defaultValue{"common", "language", "fi"},
+		defaultValue{"selfupdate", "disabled", strconv.FormatBool(false)},
+	}
+	for _, defaultValue := range defaults {
+		setIfMissing(defaultValue.section, defaultValue.key, defaultValue.value)
+	}
+}
+
+func getBoolean(section string, key string) bool {
+	value, err := getValue(section, key).Bool()
+	if err != nil {
+		mebroutines.LogDebug(fmt.Sprintf("Parsing key %s / %s as bool failed", section, key))
+		panic(fmt.Sprintf("Invalid boolean configuration flag! section: %s, key: %s", section, key))
+	}
+	return value
+}
+
+func getValue(section string, key string) *ini.Key {
```

Comment:
Tuli vielä mieleen että tässä voisi itseasiassa olla kiva jos tarkistettaisiin että haettu `section` + `key` pari löytyy defaultseista ja jos ei ole niin panikoidaan. Ehkä dollarifeaturea jatkoa ajatellen, mutta tuolla löytäisi devaajan näppäilyvirheet nopeasti ja tämä tarjoisi vähän samaa turvaa kuin alla oleva approach.

Toinen vaihtoehto olisi tyypitetty api jossa käytetyille `section`+`key` olisi constantit ja kaikki muu koodi käyttäisi niitä ja ne konvertoitaisiin to/from stringistä levyoperaatioiden yhteydessä.. ite tykkään aina tästä approachista enempi, mutta API on jo funktiopohjainen joten eiköhän mennä sillä.